app/models/segment_history.rb

```diff
@@ -22,6 +22,41 @@ def self.with_ends
     '.squish)
   end
 
+  def self.without_statistically_invalid_histories_for_run(run, timing)
```

Comment:
Can we not use `run` (i.e. `self.segment.run`) here?

Reply:
Not in a class method.

Reply:
Oh right duh. Is there any way to avoid needing to pass the run into this method?

Reply:
The join needs the Run ID, so we have to either have it be part of `Run` so we can just use `self.id` or either the run or its ID has to be passed in to whatever method is written. I also don't like passing a `Run` in since it tightly couples the 2 things together, but I think that stems from the fact these things are tightly coupled.

Reply:
Okay, that's fair.